tests/diagnostic/include.h

```diff
@@ -14,12 +14,37 @@ using namespace PHARE::diagnostic::h5;
 constexpr unsigned NEW_HI5_FILE
     = HighFive::File::ReadWrite | HighFive::File::Create | HighFive::File::Truncate;
 
+template<typename HiFile, typename Field>
+void checkField(HiFile& file, Field& field, std::string path)
+{
+    std::vector<float> fieldV;
+    file.getDataSet(path).read(fieldV);
+    EXPECT_EQ(fieldV.size(), field.size());
+
+    for (size_t i = 0; i < fieldV.size(); i++)
+    {
+        if (!std::isnan(fieldV[i]))
```

Comment:
should we check that if we have NaNs they are at expected locations? for now it's possible the test does not fail and we have all nodes at NaN

Reply:
sure, how do you want that?

Reply:
it's tricky because NaN values location/acceptance kind of depends on which field we're talking about... I think neither E or B accept NaN values. Same for the density (there is none at the moment, although we discussed that for "incomplete" nodes we might want to enforce NaN). For the fluxes I don't think we have NaNs either, same thing as for the density. For the bulk velocity component now, we saw we do have NaN at incomplete nodes for which the density is zero (so not all incomplete nodes). Those are OK. But we should not have NaNs at any other location.

Reply:
assessed